frontend/catalyst/python_interface/visualization/pydot_dag_builder.py

```
@@ -90,63 +100,78 @@ def __init__(
 
     def add_node(
```

What happens if we add a node that already exists? would it rewrite the existing node or do we want to raise an error? I know this is probably handled by `PyDot`, but would be more explicit about it or have a test.